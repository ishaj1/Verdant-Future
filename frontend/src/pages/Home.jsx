import { useState } from "react";

import LoginForm from "../components/LoginForm";
import Header from "../components/Header";
import { Link } from "react-router-dom";
import fav from "../icons/home.png";

export default function HomePage() {
  const [loginForm, setLoginForm] = useState(false);

  return (
    <>
      <Header />
      <div className="relative h-screen bg-customGreen-50">
        <div className="grid h-full max-w-screen-xl px-4 pt-20 pb-8 mx-auto lg:gap-8 xl:gap-0 lg:py-16 lg:grid-cols-12 lg:pt-28"> 
          <div class="mr-auto place-self-center lg:col-span-7">
            <h1 class="max-w-2xl mb-4 text-4xl font-extrabold leading-none tracking-tight md:text-5xl xl:text-6xl dark:text-white">Building a Verdant Future <br/>for our planet.</h1>
            <p class="max-w-2xl mb-6 font-light text-gray-500 lg:mb-8 md:text-lg lg:text-xl dark:text-gray-400">Get your company evaluated or get funds for your green project by <Link to="/register" class="hover:underline">signing up</Link>.</p>
          </div>
          <div className="m-2 lg:mt-0 lg:col-span-5 lg:flex">
            <img src={fav}  />
          </div>
        </div>
      </div>
    </>
  );
}
