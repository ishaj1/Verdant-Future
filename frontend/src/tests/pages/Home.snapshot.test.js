import React from "react";
import renderer from "react-test-renderer";
import Home from "../../pages/Home";

jest.mock("../../components/Header", () => () => <div data-testid="mock-header">Mock Header</div>);
jest.mock("../../components/LoginForm", () => () => <div data-testid="mock-login-form">Mock Login Form</div>);

describe("Home Page", () => {
  it("matches snapshot", () => {
    const tree = renderer
      .create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
