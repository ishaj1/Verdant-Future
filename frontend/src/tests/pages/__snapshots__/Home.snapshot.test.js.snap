// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
[
  <header
    aria-label="Main"
    className="flex items-center justify-between bg-customGreen-600 px-3 py-3 lg:px-12"
  >
    <a
      className="text-3xl font-bold text-white pt-5 pb-5 "
      href="/"
    >
      Verdant Future
    </a>
    <nav>
      <ul
        className="hidden gap-8 md:flex text-white font-abeezee"
      >
        <li>
          <a
            href="#"
          >
            About us
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Projects
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Companies
          </a>
        </li>
        <li>
          <button
            onClick={[Function]}
          >
            Login
          </button>
        </li>
        <li>
          <a
            className="bg-customGreen-800 px-4 py-2 rounded-xl"
            href="/register"
          >
            Sign up
          </a>
        </li>
      </ul>
      <details
        className="flex md:hidden"
      >
        <summary
          className="bg-costomGreen-600 hover:bg-customGreen-200 [[open]>&]:bg-customGreen-200 _no-triangle grid h-10 w-10 place-items-center rounded-full cursor-pointer"
        >
          <svg
            className="h-6 w-6"
            fill="none"
            stroke="white"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16m-7 6h7"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </summary>
        <div
          className="absolute right-0 z-20 md:left-0"
        >
          <div
            className="relative top-9 w-30 rounded-md border border-gray-100 bg-white p-1 shadow-sm "
          >
            <nav>
              <ul
                className="flex flex-col gap-5 px-2 py-2.5 font-abeezee text-right"
              >
                <li>
                  <a
                    href="#"
                  >
                    About us
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Projects
                  </a>
                </li>
                <li>
                  <a
                    href="#"
                  >
                    Companies
                  </a>
                </li>
                <li>
                  <button
                    onClick={[Function]}
                  >
                    Login
                  </button>
                </li>
                <li>
                  <a
                    className="bg-customGreen-800 px-4 py-2 rounded-xl text-white"
                    href="/register"
                  >
                    Sign up
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </details>
    </nav>
  </header>,
  <div
    data-testid="mock-header"
  >
    Mock Header
  </div>,
]
`;
